experimental/traffic-portal/src/app/app.component.ts

```diff
@@ -26,14 +28,15 @@ import { CurrentUserService } from "src/app/shared/currentUser/current-user.serv
 	styleUrls: ["./app.component.scss"],
 	templateUrl: "./app.component.html",
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, AfterViewInit {
+	@ViewChild("header") private readonly headerRef!: TpHeaderComponent;
```

Comment:
I kinda hate `ViewChild` because it'll compile fine if the child doesn't actually exist.


I know this is a major change to the PR, but what if instead we had a `ThemeService` or something that exposes a `BehaviorSubject` and the header just subscribes to it. Then you wouldn't need to register it either.

Reply:
The Header service now uses `ReplaySubject` instead of managing the component itself.